packages/experiments/src/components/Button/Button.styles.ts

```diff
@@ -1,10 +1,130 @@
+
+const baseTokens: IButtonComponent['tokens'] = (props, theme) => {
+  return {
+    borderRadius: 0,
+    borderWidth: 0,
+    minWidth: 100,
+    minHeight: 32,
+    lineHeight: 1,
+    contentPadding: '8px 16px',
+    textFamily: 'default',
+    textSize: 14,
+    iconSize: 14,
+    iconWeight: 400
+  };
+};
+
+const circularTokens: IButtonComponent['tokens'] = (props, theme) => {
+  return {
+    borderRadius: '50%',
+    borderWidth: 1,
+    minWidth: 32,
+    minHeight: 32,
+    contentPadding: ''
+  };
+};
+
+const enabledTokens: IButtonComponent['tokens'] = (props, theme) => {
+  const { semanticColors } = theme;
+  return {
+    backgroundColor: semanticColors.buttonBackground,
+    backgroundColorHovered: semanticColors.buttonBackgroundHovered,
+    backgroundColorPressed: semanticColors.buttonBackgroundPressed,
+
+    iconColor: semanticColors.buttonText,
+    iconColorHovered: semanticColors.buttonTextHovered,
+    iconColorPressed: semanticColors.buttonTextPressed,
+
+    color: semanticColors.buttonText,
+    colorHovered: semanticColors.buttonTextHovered,
+    colorPressed: semanticColors.buttonTextPressed,
+
+    borderColor: semanticColors.buttonBorder,
+    borderColorHovered: semanticColors.buttonBorder,
+    borderColorPressed: semanticColors.buttonBorder
+  };
+};
+
+const disabledTokens: IButtonComponent['tokens'] = (props, theme) => {
+  const { semanticColors } = theme;
+  return {
+    backgroundColor: theme.semanticColors.buttonBackgroundDisabled,
+    backgroundColorHovered: semanticColors.buttonBackgroundDisabled,
+    backgroundColorPressed: semanticColors.buttonBackgroundDisabled,
+
+    iconColor: semanticColors.buttonTextDisabled,
+    iconColorHovered: semanticColors.buttonTextDisabled,
+    iconColorPressed: semanticColors.buttonTextDisabled,
+
+    colorHovered: semanticColors.buttonTextDisabled,
+    colorPressed: semanticColors.buttonTextDisabled,
+    color: semanticColors.buttonTextDisabled,
+
+    borderColor: semanticColors.buttonBorderDisabled,
+    borderColorHovered: semanticColors.buttonBorderDisabled,
+    borderColorPressed: semanticColors.buttonBorderDisabled
+  };
+};
+
+const expandedTokens: IButtonComponent['tokens'] = (props, theme) => {
+  const { semanticColors } = theme;
+  return {
+    backgroundColor: semanticColors.buttonBackgroundPressed,
+    backgroundColorHovered: semanticColors.buttonBackgroundPressed,
+    backgroundColorPressed: semanticColors.buttonBackgroundPressed,
+
+    color: semanticColors.buttonTextPressed,
+    colorHovered: semanticColors.buttonTextPressed,
+    colorPressed: semanticColors.buttonTextPressed
+  };
+};
 
-export const getButtonStyles: IButtonComponent['styles'] = props => {
-  const { theme, disabled, expanded, className, circular, primary, styleVariables } = props;
+const primaryEnabledTokens: IButtonComponent['tokens'] = (props, theme) => {
   const { semanticColors } = theme;
+  return {
+    backgroundColor: semanticColors.primaryButtonBackground,
```

Comment:
this feels horribly redundant now. Is there a future where we don't have semanticColors but rather defaultTokens and users can access the value of that token from anywhere?

Reply:
I think it's a bit orthogonal to this PR but if we want to obsolete semanticColors and just use tokens that's an item we'll have to explore separately.

Reply:
It does feel redundant, but not to be solved in this PR. We need to work with Philip on this but I'd love to brainstorm on what this should look like. I don't think we've finalized how tokens fit in with ITheme.

Reply:
of course! not wanting to add this to the PR, just brining up the point that this is the model I always wanted for 'semantic colors' and this redundancy points out the reason why.